implants/lib/eldritch/src/file/temp_file_impl.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Result;
+use std::fs;
+use std::path::PathBuf;
+use tempfile::NamedTempFile;
+
+pub fn temp_file(name: String) -> Result<String> {
+    //create a file in temp folder
+    let tmp_file = NamedTempFile::new()?;
+    let tdir: PathBuf = tmp_file.path().parent().unwrap().into();
+    let new_path = tdir.join(name);
```

Comment:
Instead of making a temp file, and getting the parent can we use the temp dir function instead.

Rn I think this will make two temp files.